packages/devtools_app/lib/src/screens/memory/panes/diff/controller/diff_pane_controller.dart

```diff
@@ -305,9 +318,7 @@ class DerivedData extends DisposableController with AutoDisposeControllerMixin {
 
   /// Updates fields in this instance based on the values in [core].
   void _updateValues() {
-    // Make sure the method does not trigger itself recursively.
-    assert(!_updatingValues);
-    _updatingValues = true;
+    _startUpdatingValues();
```

Comment:
instead of using timeStart and timeEnd, you could use timeSync, and place the whole body of this method in the syncOperation function parameter. This will be less added lines of code and will reduce some duplication
```
ga.timeSync(
  analytics_constants.memory,     
  analytics_constants.MemoryTime.updateValues,
  syncOperation: () {
    ... the previous body of this entire method
  },
);
```

Reply:
I thought about . There are other non-functional operations to be done at start and at end. The current structure seems to be most readable for me. Yes, at cost of code duplicates.